cmd/node/main.go

```diff
@@ -1796,15 +1798,26 @@ func createNodesCoordinator(
 		return nil, nil, err
 	}
 
-	thresholdEpochDuration := epochConfig.ShuffledOutRestartThreshold
-	if !(thresholdEpochDuration >= 0.0 && thresholdEpochDuration <= 1.0) {
-		return nil, nil, fmt.Errorf("invalid threshold for shuffled out handler")
+	maxThresholdEpochDuration := epochConfig.MaxShuffledOutRestartThreshold
+	if !(maxThresholdEpochDuration >= 0.0 && maxThresholdEpochDuration <= 1.0) {
+		return nil, nil, fmt.Errorf("invalid max threshold for shuffled out handler")
 	}
-	maxDurationBeforeStopProcess := int64(nodesConfig.RoundDuration) * epochConfig.RoundsPerEpoch
-	maxDurationBeforeStopProcess = int64(thresholdEpochDuration * float64(maxDurationBeforeStopProcess))
+	minThresholdEpochDuration := epochConfig.MinShuffledOutRestartThreshold
+	if !(minThresholdEpochDuration >= 0.0 && minThresholdEpochDuration <= 1.0) {
+		return nil, nil, fmt.Errorf("invalid min threshold for shuffled out handler")
+	}
+
+	epochDuration := int64(nodesConfig.RoundDuration) * epochConfig.RoundsPerEpoch
+	minDurationBeforeStopProcess := int64(minThresholdEpochDuration * float64(epochDuration))
+	maxDurationBeforeStopProcess := int64(maxThresholdEpochDuration * float64(epochDuration))
+
+	minDurationInterval := time.Millisecond * time.Duration(minDurationBeforeStopProcess)
 	maxDurationInterval := time.Millisecond * time.Duration(maxDurationBeforeStopProcess)
-	minDurationInterval := maxDurationInterval / 2
-	//waiting interval will be [maxDuration/2 and maxDuration]
+
+	log.Debug("closing.NewShuffleOutCloser",
+		"maxDurationInterval", minDurationInterval,
```

Comment:
fix log

Reply:
you don't use minDuration else where?

Reply:
log fixed. The constructor uses both minDurationInterval and maxDurationInterval